app/src/main/java/com/example/androidtrainingproject/products/ProductDetailsViewModel.kt

```diff
@@ -0,0 +1,31 @@
+package com.example.androidtrainingproject.products
+
+import android.util.Log
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.androidtrainingproject.models.ProductResponse
+import com.example.androidtrainingproject.repository.DefaultRepository
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class ProductDetailsViewModel @Inject constructor(
+    private val repository: DefaultRepository
+): ViewModel() {
+    var productData by mutableStateOf(null as ProductResponse?)
+
+    fun getProductById(id: Number) {
+        viewModelScope.launch {
+            try {
+                val response: ProductResponse = repository.getProductById(id)
```

Comment:
Note: Unnecessary type cast - it can be inferred without specifying it. Also an unnecessary `val` - you can just straight up assign to `productData`.

Reply:
I removed that line